enhancements/network/primary-user-defined-networks-for-virtualization-workloads.md

```diff
@@ -0,0 +1,638 @@
+  - "@trozet"
+api-approvers:
+  - "@dougbtv"
+creation-date: 2024-08-30
+last-updated: 2024-09-09
+tracking-link:
+  - https://issues.redhat.com/browse/CNV-46779
+see-also:
+  - "/enhancements/network/user-defined-network-segmentation.md"
+  - "/enhancements/network/persistent-ips-sdn-secondary-networks.md"
+---
+
+# Primary User Defined Networks for Virtualization workloads
+
+## Summary
+
+Traditional virtualization users have some expectations on what a
+virtualization platform should provide. Live-migration, and IP address
+persistence across reboots are paramount features for any virtualization
+solution, and OpenShift Virtualization does not currently support any of them
+over the OpenShift default cluster network, for a variety of reasons. The scope
+of this enhancement is to define how to use the existent
+[primary UDN feature](https://github.com/openshift/enhancements/pull/1623/) to
+enable virtualization workloads to migrate over the primary network, and to
+persist the IP address across reboots or virtual machine shutdowns.
+
+## Motivation
+
+Virtualization users are commonly used to having layer2 networks, and to having
+their tenant networks isolated from other tenants. This is something that
+opposes the Kubernetes networking model, in which all pods can reach other pods,
+and security is provided by implementing Network Policy.
+
+To streamline the migration experience of users coming from these traditional
+virtualization platforms into OpenShift Virtualization we need to match the
+user's expectations in terms of feature set, and user experience. Thus, we need
+to provide them with isolated primary networks, that meet the live-migration
+and IP persistence requirements.
+
+Other type of users just want to have a more managed experience - they do not
+want to have to manage their networks, e.g. deploying DHCP servers, DNS, and
+so forth. They are hoping to have a streamlined experience, where they define
+a (set of) subnets for their layer2 network, and the SDN itself is responsible
+for assigning migratable IP addresses to the virt workloads, configure access
+to outside the cluster, and, when configured by the user, be able to access an
+application they are interested in exposing from the outside world.
+
+### User Stories
+
+- As the owner of an application running inside my VM, I want to have east/west
+communication without NAT to other VMs and/or pods within the same network.
+- As a developer who defined a custom primary network in their project, I want
+to use the primary network to which the VM is connected for north/south, while
+still being able to connect to KAPI and consume Kubernetes DNS.
+- As a VM owner, I want the VM to have the same IP address before/after live
+migrating.
+- As a VM owner, I want the VM to have the same IP address before/after restart
+or shutdown.
+- As a VM owner, I want to be able to specify the IP address for the interface
+of my VM.
```

Comment:
Maybe. As indicated in https://github.com/openshift/enhancements/pull/1673#discussion_r1753728384, we need to clarify this user story. Especially since we don't know who's the user in the story.